src/wp-includes/wp-db.php

```diff
@@ -2535,6 +2543,32 @@ protected function process_fields( $table, $data, $format ) {
 		$converted_data = $this->strip_invalid_text( $data );
 
 		if ( $data !== $converted_data ) {
+
+			$problem_fields = array();
+			foreach ( $data as $field => $value ) {
+				if ( $value !== $converted_data[ $field ] ) {
+					$problem_fields[] = $field;
+				}
+			}
+
+			if ( 1 === count( $problem_fields ) ) {
+				if ( function_exists( '__' ) ) {
+					/* translators: %s Database field where the error occurred. */
+					$message = __( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contained invalid data.' );
```

Comment:
```suggestion
					$message = __( 'WordPress database error: Processing the value for the following field failed: %s. The supplied value may be too long or contains invalid data.' );
```

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/1874/commits/3fdd2441a967146cc53ff316745219b906df8694.